src/main/resources/avro/alleleAnnotations.avdl

```diff
@@ -0,0 +1,233 @@
+@namespace("org.ga4gh.models")
+
+/**
+This protocol defines types used by the GA4GH Allele Annotation API.
+
+Annotation can be applied to the alternate alleles of classic variants
+or to Alleles
+*/
+
+
+protocol AlleleAnnotations{
+
+import idl "common.avdl";
+import idl "metadata.avdl";
+import idl "sequenceAnnotations.avdl";
+import idl "variants.avdl";
+import idl "ontologies.avdl";
+
+
+
+
+/**
+An AnalysisResult record holds the output of a prediction package such as SIFT on a
+specific allele
+*/
+record AnalysisResult {
+
+  /** The analysis record for this result (defined in metadata schema)  */
+  Analysis analysis;
+
+  /** The text-based result for this analysis */
+  union { null, string} analysisResult;
+
+  /** The numeric score for this analysis  */
+  union { null, int} analysisScore;
+
+}
+
+
+/**
+An allele location record holds the location of an allele relative to a
+non-genomic coordinate system such as a CDS or protein and holds the
+reference and alternate sequence where appropriate
+*/
+record AlleleLocation {
+
+  /** Relative start position of the allele in this coordinate system */
+  int overlapStart ;
+
+  /** Relative end position of the allele in this coordinate system  */
+  int overlapEnd ;
+
+  /** Reference sequence in feature (this should be the codon at CDS level) */
+  union { null, string } referenceSequence = null;
+
+  /** Alternate sequence in feature (this should be the codon at CDS level) */
+  union { null, string } alternateSequence = null;
+
+}
+
+/**
+Impact is a simple prioritization for the effect of an allele which is used
+in the annotation record.
+IMPORTANT:
+ Prioritization methods are a crude estimates and are not assumed to be
+ reliable: a 'HIGH' Impact may actually not cause any disruption
+ in protein function or expression.
+*/
+enum Impact {
+ HIGH,      // Variant highly disrupts protein function
+ MODERATE,  // Moderately disrupts protein function
+ LOW,       // Low disruption of protein impact
+ MODIFIER   // No known effect
+}
+
+/**
+An annotation set record groups VariantAnnotation and/or AlleleAnnotation records. It holds
+information describing the software and reference data used in the annotation.
+*/
+record AnnotationSet {
+
+  /** The ID of the annotation set record */
+  string id;
+
+  /** The ID of the dataset to which this annotation set belongs */
+  string datasetId;
+
+  /**
+  Analysis details. It is essential to supply versions for all software and
+  reference datasets used.
+  */
+  Analysis analysis;
+
+}
+
+
+/**
+A transcript effect record is a set of information describing the
+effect of an allele on a transcript
+*/
+record TranscriptEffect {
+
+
+  /** The ID of the transcript effect record
+  */
+  string id;
+
+  //TODO: derive unique id from digest of data [location, allele, transcript?]
+
+
+  /** The id of the transcript feature the annotation is relative to
```

Comment:
The TranscriptEffect record is intended just for transcripts. We will need to add other record types in future, such as one for regulatory features, but these will have different attributes.